File: Groupe2/TP2/src/bd_école.c
#include <stdio.h>
#include <string.h>

// Définition de la structure pour un étudiant
struct Etudiant {
    char nom[50];
    char prenom[50];
    char adresse[100];
    float notes[2];
};

int main() {
    struct Etudiant etudiants[5];
    int i, j;

    // Saisie des informations des étudiants
    for (i = 0; i < 5; i++) {
        printf("Etudiant %d :\n", i + 1);
        printf("Nom : ");
        fgets(etudiants[i].nom, sizeof(etudiants[i].nom), stdin);
        etudiants[i].nom[strcspn(etudiants[i].nom, "\n")] = '\0';  // Enlever '\n'

        printf("Prenom : ");
        fgets(etudiants[i].prenom, sizeof(etudiants[i].prenom), stdin);
        etudiants[i].prenom[strcspn(etudiants[i].prenom, "\n")] = '\0';

        printf("Adresse : ");
        fgets(etudiants[i].adresse, sizeof(etudiants[i].adresse), stdin);
        etudiants[i].adresse[strcspn(etudiants[i].adresse, "\n")] = '\0';

        for (j = 0; j < 2; j++) {
            printf("Note %d : ", j + 1);
            scanf("%f", &etudiants[i].notes[j]);
        }
        getchar();  // Nettoyer le buffer après scanf
        printf("\n");
    }

    // Affichage des informations des étudiants
    for (i = 0; i < 5; i++) {
        printf("Etudiant %d :\n", i + 1);
        printf("Nom : %s\n", etudiants[i].nom);
        printf("Prenom : %s\n", etudiants[i].prenom);
        printf("Adresse : %s\n", etudiants[i].adresse);
        printf("Note 1 : %.2f\n", etudiants[i].notes[0]);
        printf("Note 2 : %.2f\n", etudiants[i].notes[1]);
        printf("\n");
    }

    return 0;
}

File: Groupe2/TP2/src/bd_école.py
class Etudiant:
    def __init__(self, nom: str, prenom: str, adresse: str, notes: list[float]):
        self.nom = nom
        self.prenom = prenom
        self.adresse = adresse
        self.notes = notes

def main():
    etudiants = []

    # Saisie des informations des étudiants
    for i in range(5):
        print(f"Étudiant {i + 1} :")
        nom = input("Nom : ")
        prenom = input("Prénom : ")
        adresse = input("Adresse : ")
        notes = []
        for j in range(2):
            note = float(input(f"Note {j + 1} : "))
            notes.append(note)
        etudiants.append(Etudiant(nom, prenom, adresse, notes))
        print()

    # Affichage des informations des étudiants
    for i, etudiant in enumerate(etudiants):
        print(f"Étudiant {i + 1} :")
        print(f"Nom : {etudiant.nom}")
        print(f"Prénom : {etudiant.prenom}")
        print(f"Adresse : {etudiant.adresse}")
        print(f"Note 1 : {etudiant.notes[0]:.2f}")
        print(f"Note 2 : {etudiant.notes[1]:.2f}")
        print()

if __name__ == "__main__":
    main()
